### ISSUE TITLE
Release building broken

### ISSUE THREAD

**Issue by EdwardMoyse** [HSF/phoenix#694]
This currently fails with messages like:
```
phoenix-event-display@3.0.0 prepublishOnly /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display
> yarn build:bundle

Building Angular Packagepm pack: info lifecycle phoenix-event-display@3.0.0~prepublishOnly: phoenix-event-display@3.0.0
(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ npm pack: info lifecycle phoenix-event-display@3.0.0~prepublishOnly: phoenix-event-display@3.0.0
------------------------------------------------------------------------------
Building entry point 'phoenix-ui-components'
------------------------------------------------------------------------------
⠋ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠹ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠙ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠋ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠹ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠋ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠸ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠼ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠙ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠴ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠹ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
⠦ Compiling with Angular sources in Ivy full compilation mode.(⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂⠂) ⠹ npm pack: info lifecycle phoenix-event-display@3.0.0~prepub
✔ Compiling with Angular sources in Ivy full compilation mode.
✔ Generating FESM bundlesm pack: info lifecycle phoenix-event-display@3.0.0~prepublishOnly: phoenix-event-display@3.0.0
✔ Copying assets
⠋ Writing package manifestWARNING: Found a conflicting export condition for "./package.json". The "default" condition would be overridden by ng-packagr. Please unset it.
WARNING: Found a conflicting export condition for ".". The "types" condition would be overridden by ng-packagr. Please unset it.
WARNING: Found a conflicting export condition for ".". The "esm2022" condition would be overridden by ng-packagr. Please unset it.
WARNING: Found a conflicting export condition for ".". The "esm" condition would be overridden by ng-packagr. Please unset it.
WARNING: Found a conflicting export condition for ".". The "default" condition would be overridden by ng-packagr. Please unset it.
ℹ Removing scripts section in package.json as it's considered a potential security vulnerability.
ℹ Removing devDependencies section in package.json.
✔ Writing package manifest
✔ Built phoenix-ui-components

------------------------------------------------------------------------------
Built Angular Package
 - from: /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-ng/projects/phoenix-ui-components
 - to:   /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-ng/projects/phoenix-ui-components/dist
------------------------------------------------------------------------------

Build at: 2024-11-11T09:48:53.035Z - Time: 3146ms

assets by status 5.57 KiB [cached] 1 asseted packages/phoenix-ng/projects/phoenix-ui-components
orphan modules 42 bytes [orphan] 1 module
./src/browser.ts + 1 modules 1.95 KiB [not cacheable] [built] [code generated]

ERROR in ./src/browser.ts 2:0-50
Module not found: Error: Can't resolve './event-display.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'

ERROR in ./src/browser.ts 3:0-48
Module not found: Error: Can't resolve './lib/models/cut.model.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'

ERROR in ./src/browser.ts 4:0-63
Module not found: Error: Can't resolve './lib/models/preset-view.model.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'

ERROR in ./src/browser.ts 5:0-60
Module not found: Error: Can't resolve './helpers/pretty-symbols.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'
etc
```

I believe the ERRORs are related to `webpack` (not sure about the WARNINGs).
```
(base) emoyse@MacBookPro phoenix-event-display % yarn webpack -c configs/webpack.conf.js --stats-error-details
```

Running with `--stats-error-details` gives more information:
```
ERROR in ./src/browser.ts 22:0-90
Module not found: Error: Can't resolve './managers/ui-manager/phoenix-menu/phoenix-menu-node.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'
resolve './managers/ui-manager/phoenix-menu/phoenix-menu-node.js' in '/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src'
  using description file: /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/package.json (relative path: ./src)
    Field 'browser' doesn't contain a valid alias configuration
    using description file: /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/package.json (relative path: ./src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js doesn't exist
      .ts
        Field 'browser' doesn't contain a valid alias configuration
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js.ts doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js.js doesn't exist
      .json
        Field 'browser' doesn't contain a valid alias configuration
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js.json doesn't exist
      .wasm
        Field 'browser' doesn't contain a valid alias configuration
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js.wasm doesn't exist
      as directory
        /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js doesn't exist
```

So I don't understand `        Field 'browser' doesn't contain a valid alias configuration
` but I *think* the problem is that the path is wrong i.e.
* ` /Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/src/managers/ui-manager/phoenix-menu/phoenix-menu-node.js` doesn't exist, but:
* `/Users/emoyse/LocalDocuments/phoenix/packages/phoenix-event-display/dist/managers/ui-manager/phoenix-menu/phoenix-menu-node.js` does.

**Comment by emiliocortina** [comment]
I'll take a look this week (probably the earliest I can do is Friday). I was having some issues with my local setup too so I'll try it all over again

**Comment by EdwardMoyse** [comment]
Thanks! I'll keep investigating too. Out of interest, what are the problems you're having locally? (In case I've seen the same).

**Comment by EdwardMoyse** [comment]
@emiliocortina @9inpachi sorry to pester you, but it would really be good to be able to push all these recent improvements into a new release.

**Comment by 9inpachi** [comment]
@EdwardMoyse I have pushed to the branch you mentioned and [made a release](https://github.com/HSF/phoenix/actions/runs/11944095892/job/33294408589) from it. It should be fine now. I think running upgrade with the "application" type was causing an issue because then the libraries were being built for node instead of the browser which is wrong.

You can check the commits on https://github.com/HSF/phoenix/tree/main-new-release-builder for more details.

Please be sure to merge that branch  to mainif everything looks good since it's already published and deployed.

**Comment by EdwardMoyse** [comment]
Incredible! Thank you so much @9inpachi

**Comment by EdwardMoyse** [comment]
Hmm. I see a bunch of errors/warnings when I ran the release build action: https://github.com/HSF/phoenix/actions

I *think* it's all okay? I'm trying to update PhoenixATLAS now.

**Comment by EdwardMoyse** [comment]
Hmm ... maybe not?
```
(base) emoyse@MacBookPro PhoenixATLAS % yarn start
yarn run v1.22.22
$ ng serve
Application bundle generation failed. [2.557 seconds]

✘ [ERROR] TS2307: Cannot find module 'src/managers/ui-manager/phoenix-menu/config-types' or its corresponding type declarations. [plugin angular-compiler]
```

**Comment by EdwardMoyse** [comment]
Yeah, I just tried locally to make `3.0.1` (3.0.0 was done with the action) and it isn't working:
```
lerna info git Pushing tags...
lerna ERR! Error: Command failed with exit code 1: git push --follow-tags --no-verify --atomic origin main
lerna ERR! error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
lerna ERR! send-pack: unexpected disconnect while reading sideband packet
lerna ERR! fatal: the remote end hung up unexpectedly
lerna ERR! Everything up-to-date
lerna ERR!     at makeError (/Users/emoyse/LocalDocuments/phoenix/node_modules/execa/lib/error.js:59:11)
lerna ERR!     at handlePromise (/Users/emoyse/LocalDocuments/phoenix/node_modules/execa/index.js:114:26)
lerna ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
lerna ERR!     at async Promise.all (index 0)
lerna ERR! Error: Command failed with exit code 1: git push --follow-tags --no-verify --atomic origin main
lerna ERR! error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
lerna ERR! send-pack: unexpected disconnect while reading sideband packet
lerna ERR! fatal: the remote end hung up unexpectedly
lerna ERR! Everything up-to-date
lerna ERR!     at makeError (/Users/emoyse/LocalDocuments/phoenix/node_modules/execa/lib/error.js:59:11)
lerna ERR!     at handlePromise (/Users/emoyse/LocalDocuments/phoenix/node_modules/execa/index.js:114:26)
lerna ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
lerna ERR!     at async Promise.all (index 0)
lerna ERR! lerna Command failed with exit code 1: git push --follow-tags --no-verify --atomic origin main
lerna ERR! lerna error: RPC failed; HTTP 400 curl 22 The requested URL returned error: 400
lerna ERR! lerna send-pack: unexpected disconnect while reading sideband packet
lerna ERR! lerna fatal: the remote end hung up unexpectedly
lerna ERR! lerna Everything up-to-date
```

**Comment by EdwardMoyse** [comment]
Trying with `2.1.7`, I see:
```
./node_modules/phoenix-event-display/dist/loaders/phoenix-loader.js:10:0-28 - Error: Module not found: Error: Can't resolve 'lodash' in '/Users/emoyse/LocalDocuments/PhoenixATLAS/node_modules/phoenix-event-display/dist/loaders'

Error: node_modules/phoenix-event-display/dist/lib/models/cut.model.d.ts:1:35 - error TS2307: Cannot find module 'src/managers/ui-manager/phoenix-menu/config-types' or its corresponding type declarations.

1 import { ConfigRangeSlider } from 'src/managers/ui-manager/phoenix-menu/config-types';
```
So actually, it might be `lodash` which is the issue. 

Fixed in `3.0.2` but I still see:
```
Error: node_modules/phoenix-event-display/dist/lib/models/cut.model.d.ts:1:35 - error TS2307: Cannot find module 'src/managers/ui-manager/phoenix-menu/config-types' or its corresponding type declarations.

1 import { ConfigRangeSlider } from 'src/managers/ui-manager/phoenix-menu/config-types';
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
But unless this is just the first of many, seems like that the release building IS working and this is really a specific issue with `config-types`
